### ISSUE TITLE
waitForNavigation should wait for all frames instead of only main frame

### ISSUE THREAD

**Issue by yujunglo** [puppeteer/puppeteer#1173]
- Platform/OS: macOS High Sierra
- Puppeteer version: Puppeteer version: ab9b34cd5dad54e170be916016a4e9b4097231ed (from master branch as of Oct 24), 0.12.0 has same problem

Take this url http://diy-cellars.space/woodworking-plans-glider/?DIYCellar=301248662559422964
 and the following code snippet for example.
```

async function dumpFrameTree(page, frame, indent) {
    console.log(indent + frame.url());
    const result = await frame.evaluate(() => {
        let retVal = '';
        if (document.doctype) {
            retVal = new XMLSerializer().serializeToString(document.doctype);
        }
        if (document.documentElement) {
            retVal += document.documentElement.outerHTML;
        }
        return retVal;
    });
    console.log(indent + "  " + result.slice(0, 20));
    for (let child of frame.childFrames()) {
        await dumpFrameTree(page, child, indent + '  ');
    }
}

await page.goto(url, {waitUntil: 'networkidle0', timeout: 60000});
await dumpFrameTree(page, page.mainFrame(), '');
```

The above code will print:
```
http://diy-cellars.space/woodworking-plans-glider/?DIYCellar=301248662559422964
  <html><head></head><
  http://diy-cellars.space/pin.php?ProcessId=301248662559422964
    <html><head><script 
  
TypeError: Cannot read property 'evaluate' of null
```

If sleep for 5 seconds between `page.goto` and `dumpFrameTree`, we can get the expected result:
```
http://diy-cellars.space/woodworking-plans-glider/?DIYCellar=301248662559422964
  <html><head></head><
  http://diy-cellars.space/pin.php?ProcessId=301248662559422964
    <html><head><script 
  http://www.tedswoodworking.com/new/main/
    <!DOCTYPE html><html
    https://player.vimeo.com/video/71084479
      <!DOCTYPE html><html
    https://player.vimeo.com/video/71084478
      <!DOCTYPE html><html
```

The waitForNavigation function that page.goto delegates to only listen to main frame event instead of all frames. If goto listens to all the frame event (or that there's a workaround), this will be addressed.

**Comment by aslushnikov** [comment]
@yujunglo good catch. That's indeed the problem with 0.13: while migrating to lifecycle events, we should make sure all the frames in the subtree reached the desired state as well. I'll fix this.

However, it should work fine with `waitUntil: 'networkidle'` argument with puppeteer 0.12. The following worked just fine for me:

```js
const puppeteer = require('puppeteer');

(async () => {
  async function dumpFrameTree(page, frame, indent) {
      console.log(indent + frame.url());
      const result = await frame.evaluate(() => {
          let retVal = '';
          if (document.doctype) {
              retVal = new XMLSerializer().serializeToString(document.doctype);
          }
          if (document.documentElement) {
              retVal += document.documentElement.outerHTML;
          }
          return retVal;
      });
      console.log(indent + "  " + result.slice(0, 20));
      for (let child of frame.childFrames()) {
          await dumpFrameTree(page, child, indent + '  ');
      }
  }

  const browser = await puppeteer.launch();
  const page = await browser.newPage();
  const url = 'http://diy-cellars.space/woodworking-plans-glider/?DIYCellar=301248662559422964';
  await page.goto(url, {waitUntil: 'networkidle', timeout: 60000});
  await dumpFrameTree(page, page.mainFrame(), '');
  await browser.close();
})();

```

**Comment by yujunglo** [comment]
Confirmed that this works with puppeteer 0.12. Look forward to the fix in 0.13. Thank you.

**Comment by udaykanthr** [comment]
Hi ,

We are using latest version instead of 0.12.
When can we expect this fix applied.
